### ISSUE TITLE
cmake : build bitpit as a subproject

### ISSUE THREAD

**Issue by pkestene** [optimad/bitpit#37]
Hello,

I would like to suggest the following minor change in the top-level CMakeLists.txt in order to enable bitpit to be built as a subproject of another cmake project, e.g. like this:

```cmake
project(OUTER_PROJ)
...
add_subdirectory(bitpit)
```

This currently doesn't work, but very tiny cmake modifications are required to re-enable this:

1. turn CMAKE_PROJECT_NAME into PROJECT_NAME
Indeed, CMAKE_PROJECT_NAME expands to the outer project name (which is not desirable), while PROJECT_NAME expands to the nearest project command (which is what we want)
2. the way CMAKE_MODULE_PATH is set should CMAKE_CURRENT_SOURCE_DIR instead of CMAKE_SOURCE_DIR for the same reason that CMAKE_SOURCE_DIR expands to the top-level cmake project command

Let me know if you agree on this. I can provide a PR for it if needed.

**Comment by andrea-iob** [comment]
The changes seems fine. Please do separate commits for the two changes. Thanks.

Do you know in which version of cmake the two commands PROJECT_NAME and CMAKE_CURRENT_SOURCE_DIR were introduced? bitpit currently can be compiled with cmake 2.8.6. If the required commands where introduced later, we may think about increasing the minimum required version of cmake (2.8.6 is quite old).

**Comment by pkestene** [comment]
I check cmake doc for [version 2.8.6](https://cmake.org/cmake/help/v2.8.6/cmake.html) both variables are defined; I also cross-check by recompiling this old version, it's ok.

Please note, there are actually multiple occurences of CMAKE_PROJECT_NAME that can be safely replaced by PROJECT_NAME; indeed since command "project" is only called once in bitpit, they are equal, it is only in the case bitpit is built as part as of another outper project that the problem pops up.

**Comment by pkestene** [comment]
I made the changes mentionned in https://github.com/pkestene/bitpit/tree/issue37

I think it is a good practice to favor PROJECT_NAME instead of CMAKE_PROJECT_NAME
but  for what i wanted to do, i finaly realized it was probably a bad idea to build bitpit using add_subdirectory(bitbit); it is a kind of dirty trick because merges both build systems, the one of the external dependency with the main project. I discovered it is dirty just because for example, the external depency can define custom targets e.g. "add_custom_target(check)" which may conflict with other custom target defined in the main project with the same name.

I'll finaly use cmake command "ExternalProject_Add" instead of "add_directory".
